packages/grid/_modules_/grid/hooks/root/useRows.ts

```diff
@@ -30,7 +30,7 @@ export const useRows = (
   apiRef: ApiRef,
 ): RowModel[] => {
   const logger = useLogger('useRows');
-  const rowModels = React.useMemo(() => rows.map((r) => createRow(r)), [rows]);
+  const rowModels = React.useMemo(() => rows.map((r, idx) => createRow(r, idx)), [rows]);
```

Comment:
Be aware of:  https://ag-grid.com/javascript-grid-infinite-scrolling/#selection

> Selection works on the rows in infinite scrolling by using the ID of the row node. If you do not provide IDs for the row nodes, the index of the row node will be used. Using the index of the row breaks down when (server-side) filtering or sorting, as these change the index of the rows. For this reason, if you do not provide your own IDs, then selection is cleared if sort or filter is changed.

If we go down the road of allowing rows without ID, there will be work later on. As you as you are happy with the implication, I'm happy too.

Reply:
I know that's why I didn't support rows without id originally. 
But javascript user can bypass the ts constraint, so I can throw an error if a row doesn't have an id or a warning and use the index. In the doc we can put a warning as they did.